_test/general.test.php

```diff
@@ -35,27 +95,364 @@ public function test_plugininfo() {
+        $attributeKey = (string)$attributes[0]->key[0];
+        $attributeValue = (string)$attributes[0]->values[0]->value[0];
+
+        $dateAddeds = $xml->xpath('/findologic/items/item/dateAddeds');
+        $dateAdded = (string)$dateAddeds[0]->dateAdded[0];
+
+        $expectedName = $pageTitle;
+        $expectedSummary = $expectedDescription = self::PAGE_CONTENT_PLACEHOLDER;
+        $expectedOrdernumber = $pageId;
+        $expectedUrl = 'http://wiki.example.com/./doku.php?id=' . $pageId;
+        $expectedPropertyKey = 'dummy';
+        $expectedPropertyValue = 'dummy';
+        $expectedAttributeKey = 'cat';
+        $expectedAttributeValue = 'test123_test123_test123';
+        // Format DateTime because the creation date is not known. It can vary.
+        $pageMetadata = p_get_metadata($pageId);
+        $pageCreated = new DateTime();
+        $pageCreated->setTimestamp($pageMetadata["date"]["created"]);
+        $expectedDateAdded = (string)$pageCreated->format(\DateTime::ATOM);
+
+        $this->assertEquals($expectedName, $name, 'Expected name in XML should match the pages first title "test123".');
+        $this->assertEquals($expectedSummary, $summary, 'Expected summary in XML should match the pages content.');
+        $this->assertEquals($expectedDescription, $description, 'Expected description in XML should match the pages content.');
+        $this->assertEquals($expectedOrdernumber, $ordernumber, 'Expected ordernumber in XML should match the pages namespace "test123:test123:test123".');
+        $this->assertEquals($expectedUrl, $url, 'Expected url in XML should match the DokuWiki-URL and the pages namespace "http://wiki.example.com/./doku.php?id=test123:test123:test123".');
+        $this->assertEquals($expectedPropertyKey, $propertyKey, 'Expected property key in XML should match the dummy value "dummy".');
+        $this->assertEquals($expectedPropertyValue, $propertyValue, 'Expected property value in XML should match the dummy value "dummy".');
+        $this->assertEquals($expectedAttributeKey, $attributeKey, 'Expected attribute key in XML should match the category value "cat".');
+        $this->assertEquals($expectedAttributeValue, $attributeValue, 'Expected attribute value in XML should match the namespace formatted in a FINDOLOGIC proper format "test123_test123_test123".');
+        $this->assertEquals($expectedDateAdded, $dateAdded, 'Expected dateAdded value in XML should match the created date of the page. Value can vary.');
+    }
+
+    /**
+     * Test to ensure that if no title is given, the XML will have the namespace well formated as title.
+     */
+    public function test_xml_element_name_is_formatted_namespace_from_dokuwiki_page_data()
+    {
+        $this->savePages('test321:test321:test321');
+
+        $xml = $this->getXML();
+
+        $names = $xml->xpath('/findologic/items/item/names/name');
+        $name = (string)$names[0];
+
+        $expectedName = 'Test321 Test321 Test321';
```

Comment:
Test the current behavior so we can merge the changes, and create an issue if we want to change the behavior.

Reply:
Alright I will check for an empty string.